gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -89,10 +89,13 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
    * @param namespace namespace of the table this extract belongs to
    * @param table name of the table this extract belongs to
    * @return a new unique {@link Extract} instance
+   *
+   * @Deprecated Use {@link Extract#getUniqueInstance(Extract.TableType, String, String)}
```

Comment:
For now, you can also make EXTRACT_SET and ExtractFactoryImpl and reuse the implementation in createExtract to avoid code duplication.